pkgs/development/python-modules/scipy/default.nix

```diff
@@ -55,10 +55,12 @@ let
       sha256 = hash;
     }
   ) datasetsHashes;
+  isNumpy2 = (lib.versions.major numpy.version) == "2";
+  numpyCoreDir = "${lib.optionalString isNumpy2 "_"}core";
```

Comment:
When numpy 1 and 2 coexist, it is hard to do such a hack for every downstream package.
Could we add `dev` to outputs to integrate the differences?

Reply:
> When numpy 1 and 2 coexist, it is hard to do such a hack for every downstream package. Could we add `dev` to outputs to integrate the differences?

Hmm I don't think stdenv will detect the headers in that location and move them automatically to `$dev/include` or something alike, so `moveToOutput` will have to be used. Then would you move it to `$dev/include`? or to `$dev/${python.sitePackages}/numpy/${coreDir}/include`? In the later case what benefit that would bring? It'd even be less trivial for packages to guess that path (in case they do)? Are we even aware of packages that try to detect that path dynamically during their build? I'm afraid that a `$dev` output can only bother them..

We could add that `coreDir` as a `passthru` variable for both `numpy_{1,2}` which will make this feel less of a hack. What do you think?